### PR TITLE
Use rubocop auto correct to fix a number of issues

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?
Helps with #6055 

Fix rubocop issues with auto correct.
Also update rubocop todo lists so rubocop runs without warnings (there were +100 issues not listed in the todo lists).

#### What should we test?
This PR is 100% the result of running rubocop -a and most of changes are in specs so with a green build this is safe to merge.

#### Release notes
Changelog Category: Technical changes
Improve code style with rubocop.

### CODE REVIEW

File: spec/factories/stock_location_factory.rb [hunk 1]
```diff
@@ -20,14 +20,14 @@
...
+      after(:create) do |stock_location, _evaluator|
         # variant will add itself to all stock_locations in an after_create
         # creating a product will automatically create a master variant
         product_1 = create(:product)
         product_2 = create(:product)
 
-        stock_location.stock_items.where(:variant_id => product_1.master.id).first.adjust_count_on_hand(10)
-        stock_location.stock_items.where(:variant_id => product_2.master.id).first.adjust_count_on_hand(20)
+        stock_location.stock_items.where(variant_id: product_1.master.id).first.adjust_count_on_hand(10)
+        stock_location.stock_items.where(variant_id: product_2.master.id).first.adjust_count_on_hand(20)
```
Comment (andrewpbrett) [inline spec/factories/stock_location_factory.rb:30]: haha no more #🚀👍

File: spec/models/spree/credit_card_spec.rb [hunk 2]
```diff
@@ -31,8 +31,8 @@ def stub_rails_env(environment)
 
         it "should be true if payment is checkout" do
           payment = build_stubbed(:payment, created_at: Time.zone.now)
-          allow(payment).to receive_messages :pending? => false,
-                                             :checkout? => true
+          allow(payment).to receive_messages pending?: false,
+                                             checkout?: true
```
Comment (andrewpbrett) [inline spec/models/spree/credit_card_spec.rb:35]: having the `?` at the end of the key looks odd but it's valid so 👍

File: lib/spree/core/controller_helpers/auth.rb [hunk 3]
```diff
@@ -43,7 +43,7 @@ def store_location
             end
           end
 
-          disallowed_urls.map!{ |url| url[/\/\w+$/] }
+          disallowed_urls.map!{ |url| url[%r{/\w+$}] }
```
Comment (andrewpbrett) [inline lib/spree/core/controller_helpers/auth.rb:46]: This is just replacing the regex `/.../` notation with the `%r{...}` notation, right?

**Review by andrewpbrett — APPROVED** [review]
This looks good to me, lots of whitespace/alignment fixes and hashrocket-ectomies. 👍

File: spec/models/spree/credit_card_spec.rb (hunk 2 above)
Comment (luisramos0) [reply spec/models/spree/credit_card_spec.rb:35]: yes, these are methods from state_machine to check if the state is in a given point like pending? these will only be keys here where we mock the calls to these methods.

File: lib/spree/core/controller_helpers/auth.rb (hunk 3 above)
Comment (luisramos0) [reply lib/spree/core/controller_helpers/auth.rb:46]: yes, correct :+1: 
thanks for the review!

**Review by sauloperez — APPROVED** [review]

**Pull request merged**